### PR TITLE
Feature/master/add check for incorrect path

### PR DESCRIPTION
**Pull request by thomashohn**
Fixes: #967

### CODE REVIEW

File: Classes/Backend/SolrModule/OverviewModuleController.php [hunk 1]
```diff
@@ -76,6 +76,7 @@ protected function checkConnections()
     {
         $connectedHosts = [];
         $missingHosts = [];
+        $invalidPath = [];
```
Comment (irnnr) [inline Classes/Backend/SolrModule/OverviewModuleController.php]: should be plural

File: Classes/Backend/SolrModule/OverviewModuleController.php [hunk 2]
```diff
@@ -85,6 +86,11 @@ protected function checkConnections()
             } else {
                 $missingHosts[] = $coreUrl;
             }
+
+            // Check that path is valid
+            if ($this->validatePath($connection->getPath()) === false) {
```
Comment (irnnr) [inline Classes/Backend/SolrModule/OverviewModuleController.php]: `if (!$this->isValidPath($connection->getPath())) {`

File: Classes/Backend/SolrModule/OverviewModuleController.php [hunk 3]
```diff
@@ -102,6 +108,32 @@ protected function checkConnections()
...
+    }
+
+    /**
+     * Validate solr path
+     *
+     * @param string $path
+     *
+     * @return bool
+     */
+    protected function validatePath($path) {
```
Comment (irnnr) [inline Classes/Backend/SolrModule/OverviewModuleController.php]: `protected function isValidPath($path) {`

File: Classes/Backend/SolrModule/OverviewModuleController.php [hunk 4]
```diff
@@ -102,6 +108,32 @@ protected function checkConnections()
...
+            $this->addFlashMessage(
+                'Path should not contain the characters "*, ?, <, >, |, :, or #":' . PHP_EOL . implode(PHP_EOL, $invalidPath),
+                'Path is not valid',
+                FlashMessage::WARNING
+            );
+        }
+    }
+
+    /**
+     * Validate solr path
```
Comment (irnnr) [inline Classes/Backend/SolrModule/OverviewModuleController.php]: Solr with uppercase S

File: Classes/Backend/SolrModule/OverviewModuleController.php [hunk 5]
```diff
@@ -102,6 +108,32 @@ protected function checkConnections()
...
+     *
+     * @param string $path
+     *
+     * @return bool
+     */
+    protected function validatePath($path) {
+        $path = trim($path);
+
+        // Check for invalid characters
+        if(preg_match('/^[^*?"<>|:#]*$/',$path)) {
```
Comment (irnnr) [inline Classes/Backend/SolrModule/OverviewModuleController.php]: `if (`

File: Classes/Backend/SolrModule/OverviewModuleController.php [hunk 6]
```diff
@@ -85,6 +86,11 @@ protected function checkConnections()
             } else {
                 $missingHosts[] = $coreUrl;
             }
+
+            // Check that path is valid
```
Comment (irnnr) [inline Classes/Backend/SolrModule/OverviewModuleController.php]: unneeded comment, this is obvious

**Comment by thomashohn** [comment]
@timohund I can do that - so a class called PathUtility under Utility or?

**Comment by timohund** [comment]
@thomashohn I would propose "System/Validator/Path"

**Pull request merged**